src/js/mqtt.js

```diff
@@ -105,10 +110,19 @@ MqttClient.prototype.connect = function() {
   return this;
 };
 
+MqttClient.prototype._onConnectTimeout = function() {
+  this.emit('error', new Error('connect timeout'));
+  this._ondisconnect();
+};
+
 /**
  * @method _onconnect
  */
 MqttClient.prototype._onconnect = function() {
+  if (this._connectTimer) {
+    clearTimeout(this._connectTimer);
+    this._connectTimer = null;
+  }
```

Comment:
See https://github.com/mqttjs/MQTT.js#mqttclientstreambuilder-options, we should reset the state when receiving the CONNACK packet.